### PR TITLE
Add sim:toywasm config

### PR DESCRIPTION
**Pull request by yamt**
## Summary
Add sim:toywasm config

## Testing
Tested on macOS.

### CODE REVIEW

File: boards/sim/sim/sim/README.txt [hunk 1]
```diff
@@ -1245,6 +1245,18 @@ touchscreen
...
+  This is a configuration with toywasm.
+
+  An example usage:
+
+    NuttShell (NSH) NuttX-10.4.0
+    nsh> mount -t hostfs -o fs=/tmp/wasm /mnt
+    nsh> toywasm --wasi /mnt/hello.wasm
+    hello
+    nsh>
+
```
Comment (acassis) [inline boards/sim/sim/sim/README.txt:1259]: Please move to Documentation/
We are trying to phase out all boards README.txt and keep everything in the web site.

Comment (yamt) [reply boards/sim/sim/sim/README.txt:1259]: do you mean to block this PR?

**Comment by xiaoxiang781216** [comment]
@yamt please resolve the conflict.

**Comment by yamt** [comment]
> @yamt please resolve the conflict.

done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**